plutus-core/plutus-core/test/Value/Spec.hs

```diff
@@ -34,19 +33,21 @@ prop_packPreservesInvariants = forAll arbitrary $ \nm ->
 
 -- | Verifies that @insertCoin@ correctly updates the sizes
 prop_insertCoinBookkeeping :: Property
-prop_insertCoinBookkeeping = forAll arbitrary $ \(v, amt) ->
+prop_insertCoinBookkeeping = forAll arbitrary $ \v ->
   forAll (genShortHex (V.totalSize v)) $ \currency ->
     forAll (genShortHex (V.totalSize v)) $ \token ->
-      let v' = V.insertCoin currency token amt v
-       in checkSizes v'
+      forAll arbitraryBuiltin $ \amt ->
+        let v' = V.insertCoin currency token amt v
+         in checkSizes v'
 
 -- | Verifies that @insertCoin@ preserves @Value@ invariants
 prop_insertCoinPreservesInvariants :: Property
-prop_insertCoinPreservesInvariants = forAll arbitrary $ \(v, amt) ->
+prop_insertCoinPreservesInvariants = forAll arbitrary $ \v ->
   forAll (genShortHex (V.totalSize v)) $ \currency ->
     forAll (genShortHex (V.totalSize v)) $ \token ->
-      let v' = V.insertCoin currency token amt v
-       in checkInvariants v'
+      forAll arbitraryBuiltin $ \amt ->
```

Comment:
Don't use `forAll` unless you absolutely have to, it's another footgun helpfully provided by QuickCheck with no commentary. It doesn't perform any shrinking. Just bind `v` and `amt` directly:

```haskell
prop_insertCoinPreservesInvariants :: Value -> Integer -> Property
prop_insertCoinPreservesInvariants = <...>
```

Alternatively, you can use `forAllShrink`, but IIRC that makes shrinking sequential, i.e. once shrinking the first argument stops failing the test, you start shrinking the second and never come back to try shrinking the first again.

Reply:
>  Just bind v and amt directly

Does that use `arbitrary @Integer`? If so does that mean I'll need to wrap it in a newtype?

Reply:
Oh yes, sorry, I'm stupid, you need the newtype wrapper.